### ISSUE TITLE
directory creation for buildquiz

### ISSUE THREAD

**Issue by dcsan** [mistval/kotoba#102]
```
➜  bot git:(master) npm run buildquiz

> kotoba@1.0.0 buildquiz /Users/dc/dev/rnd/kotoba/bot
> node ./src/build/build_quiz_decks.js

Building quiz data
(node:7055) ExperimentalWarning: The fs.promises API is experimental
{ [Error: ENOENT: no such file or directory, mkdir '/Users/dc/dev/rnd/kotoba/bot/generated/quiz/decks']
  errno: -2,
  code: 'ENOENT',
  syscall: 'mkdir',
  path: '/Users/dc/dev/rnd/kotoba/bot/generated/quiz/decks' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! kotoba@1.0.0 buildquiz: `node ./src/build/build_quiz_decks.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the kotoba@1.0.0 buildquiz script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/dc/.npm/_logs/2019-08-13T08_17_43_346Z-debug.log

➜  bot git:(master) mkdir -p generated/quiz/decks
➜  bot git:(master) npm run buildquiz

> kotoba@1.0.0 buildquiz /Users/dc/dev/rnd/kotoba/bot
> node ./src/build/build_quiz_decks.js

Building quiz data
(node:7125) ExperimentalWarning: The fs.promises API is experimental
{ [Error: EEXIST: file already exists, mkdir '/Users/dc/dev/rnd/kotoba/bot/generated/quiz/decks']
  errno: -17,
  code: 'EEXIST',
  syscall: 'mkdir',
  path: '/Users/dc/dev/rnd/kotoba/bot/generated/quiz/decks' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! kotoba@1.0.0 buildquiz: `node ./src/build/build_quiz_decks.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the kotoba@1.0.0 buildquiz script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/dc/.npm/_logs/2019-08-13T08_18_18_657Z-debug.log
➜  bot git:(master) pwd
/Users/dc/dev/rnd/kotoba/bot

➜  bot git:(master) rmdir generated/quiz/decks
➜  bot git:(master) npm run buildquiz

> kotoba@1.0.0 buildquiz /Users/dc/dev/rnd/kotoba/bot
> node ./src/build/build_quiz_decks.js

Building quiz data
(node:7185) ExperimentalWarning: The fs.promises API is experimental
done
```

it seems this needs
- mkdir -p to create the path to the dir `generated/quiz` so `/decks` can be added
- check if the dir exists first to avoid `EEXIST`

**Comment by mistval** [comment]
That's weird, I thought the new recursive option in fs.mkdir was supposed to not error if directory exists 🤔 recently I replaced mkdirp with `fs.mkdir(path, { recursive: true })` since that's a thing now. Will take a look later.

**Comment by mistval** [comment]
Ooooh fs.promises.mkdir doesn't support the recursive option yet, rip. I'll just make that synchronous then.

**Comment by mistval** [comment]
Actually it does 🤔 I misread. So now I'm confused again.

**Comment by dcsan** [comment]
related. this seemed to fix itself with node 10.12 I was on 10.10 before.

```
➜  bot git:(master) ✗ yarn start
yarn run v1.17.3
$ node --max_old_space_size=2000 .
fs.js:115
    throw err;
    ^

Error: EEXIST: file already exists, mkdir '/Users/dc/dev/rnd/kotoba/bot/data'
    at Object.mkdirSync (fs.js:731:3)
    at createBot (/Users/dc/dev/rnd/kotoba/bot/src/start.js:11:6)
```
